src/mcp/validation.ts

```
@@ -1,5 +1,26 @@
 import { z } from 'zod';
 
+const FileNameSchema = z.string()
+  .min(1, 'Filename is required')
+  .max(255, 'Filename too long')
+  .refine(
+    (name) => {
+      // Path separators and traversal
+      if (/[\\\/]/.test(name)) return false;
+      if (/\.\./.test(name)) return false;
+      // Control characters (0x00-0x1F, 0x7F)
+      if (/[\x00-\x1F\x7F]/.test(name)) return false;
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)